src/lib/transaction_snark/test/zkapp_fuzzy/zkapp_fuzzy.ml

```diff
@@ -91,177 +92,285 @@ let mk_ledgers_and_fee_payers ?(is_timed = false) ~num_of_fee_payers () =
   in
   (ledger, fee_payer_keypairs, keymap)
 
-let generate_zkapp_commands_and_apply_them_consecutively ~trials
-    ~max_account_updates () =
+let generate_zkapp_commands_and_apply_them_consecutively_5_times ~successful
+    ~max_account_updates ~individual_test_timeout random =
   let ledger, fee_payer_keypairs, keymap =
-    mk_ledgers_and_fee_payers ~num_of_fee_payers:trials ()
+    mk_ledgers_and_fee_payers ~num_of_fee_payers:5 ()
   in
   let account_state_tbl = Account_id.Table.create () in
   let global_slot = Mina_numbers.Global_slot_since_genesis.one in
-  Test_util.with_randomness 123456789 (fun () ->
-      let test i =
-        Quickcheck.test ~trials:1
-          (Mina_generators.Zkapp_command_generators.gen_zkapp_command_from
-             ~global_slot ~protocol_state_view:U.genesis_state_view
-             ~account_state_tbl ~fee_payer_keypair:fee_payer_keypairs.(i)
-             ~max_account_updates ~keymap ~ledger ~vk () )
-          ~f:(fun zkapp_command_dummy_auths ->
-            let open Async in
-            Thread_safe.block_on_async_exn (fun () ->
-                let%bind.Deferred zkapp_command =
-                  Zkapp_command_builder.replace_authorizations ~prover ~keymap
-                    zkapp_command_dummy_auths
-                in
-                [%log info]
-                  ~metadata:
-                    [ ("zkapp_command", Zkapp_command.to_yojson zkapp_command)
-                    ; ( "accounts"
-                      , `List
-                          (List.map
-                             (Mina_ledger.Ledger.to_list_sequential ledger)
-                             ~f:Account.to_yojson ) )
-                    ]
-                  "generated zkapp_command" ;
-                U.check_zkapp_command_with_merges_exn ~global_slot ledger
-                  [ zkapp_command ] ) )
-      in
-      for i = 0 to trials - 1 do
-        test i
-      done )
+  let test i random =
+    let zkapp_command_dummy_auths =
+      Quickcheck.Generator.generate ~size:10 ~random
+        (Mina_generators.Zkapp_command_generators.gen_zkapp_command_from
+           ~global_slot ~protocol_state_view:U.genesis_state_view
+           ~account_state_tbl ~fee_payer_keypair:fee_payer_keypairs.(i)
+           ~max_account_updates ~keymap ~ledger ~vk () )
+    in
+    let%bind.Deferred zkapp_command =
+      Zkapp_command_builder.replace_authorizations ~prover ~keymap
+        zkapp_command_dummy_auths
+    in
+    let%map () =
+      match%map
+        try_with (fun () ->
+            with_timeout (Time.Span.of_int_sec individual_test_timeout)
+            @@ U.check_zkapp_command_with_merges_exn ~global_slot ledger
+                 [ zkapp_command ] )
+      with
+      | Ok (`Result ()) ->
```

Comment:
Nit: this match statement boilerplate is repeated a few times. Would be nicer to factor it out as a function.

Reply:
The match statement has some difference in the successful and failure cases. If we want to abstract this out, we also need to include the generation of the zkapp. I would leave it as it is for now.